tony/tony.sh

```diff
@@ -0,0 +1,173 @@
+#!/bin/bash
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# This script installs TonY on a master node within a Google Cloud Dataproc cluster.
+
+set -x -e
+
+readonly TONY_INSTALL_FOLDER='/usr/local/src'
+readonly TONY_SAMPLES_FOLDER="${TONY_INSTALL_FOLDER}"'/TonY-samples'
+readonly TENSORFLOW_VERSION='1.9'
+readonly PYTORCH_VERSION='0.4.1'
+
+# TonY configurations
+readonly PS_INSTANCES=1
+readonly PS_MEMORY='4g'
+readonly WORKER_INSTANCES=2
+readonly WORKER_MEMORY='4g'
+readonly WORKER_GPUS=0     # Not supported in Dataproc
+
+
+function err() {
+  echo "[$(date +'%Y-%m-%dT%H:%M:%S%z')]: $@" >&2
+  return 1
+}
+
+function download_and_build_tony() {
+  # Download TonY latest distribution.
+  cd "${TONY_INSTALL_FOLDER}"
+  git clone https://github.com/linkedin/TonY.git
+  cd TonY
+  # Build TonY without tests.
+  ./gradlew build -x test
```

Comment:
I am not sure if gradlew is available by default.

Reply:
It is available via TonY repo.